### DISCUSSION TITLE
I have problem with api

### DISCUSSION THREAD

**Discussion by BartOrd** [pufferpanel/pufferpanel#1369] [Q&A]
if i trided to create server by api i have this error:
{
  "error": {
    "msg": "undefined environment: ",
    "code": "ErrGeneric"
  }
}
can we help me?
api code:
{
  "servers": [
    {
      "name": "defaultserver",
      "node": {
        "id": 0,
        "name": "LocalNode",
        "publicHost": "panel.l6vel.xyz",
        "publicPort": 80,
        "sftpPort": 5657,
        "isLocal": true,
        "type": "node"
      },
      "ip": "0.0.0.0",
      "type": "bot",
      "os": "discord.js"
    }
  ],
  "data": {
    "bot-start-command": {
      "type": "string",
      "desc": "The bot start command.",
      "display": "Bot start command",
      "required": true,
      "value": "node ./bot.js"
    }
  },
  "requirements": {
    "binaries": [
      "node",
      "npm"
    ]
  },
  "run": {
    "stopCode": 2,
    "workingDirectory": "",
    "pre": [],
    "post": []
  },
  "type": "discord"
}
shablon i use:
{
  "name": "discord-js",
  "display": "discord.js",
  "type": "discord",
  "install": [
    {
      "commands": [
        "npm i"
      ],
      "type": "command"
    }
  ],
  "run": {
    "stopCode": 2,
    "command": "${bot-start-command}",
    "workingDirectory": "",
    "pre": [],
    "post": [],
    "environmentVars": {}
  },
  "data": {
    "bot-start-command": {
      "type": "string",
      "desc": "The bot start command.",
      "display": "Bot start command",
      "required": true,
      "value": "node ./bot.js"
    }
  },
  "environment": {
    "type": "standard"
  },
  "supportedEnvironments": [
    {
      "type": "standard"
    }
  ],
  "requirements": {
    "binaries": [
      "node",
      "npm"
    ]
  }
}

**Comment by BartOrd** [comment]
![image](https://github.com/user-attachments/assets/0b9aceca-7f54-499b-a693-dddb0cfe4d33)
![image](https://github.com/user-attachments/assets/93c184a9-b7ef-48f9-8c0a-7ac8821b9d16)

**Comment by LordRalex** [comment]
Please specify your version and also format your JSON blocks so they are readable.

**Comment by BartOrd** [comment]
you can explain more, sorry for these questions, but I'm a beginner in this
